src/tests/tags-service.test.js

```diff
@@ -0,0 +1,273 @@
+        // Add promise to remove tuits with test text to list of promises
+        promises.push(deleteTuitByTuitText(mockedTuit.tuit));
+
+        await Promise.all(promises);    // Wait for all promises to be fulfilled
+        console.log("Before user creation");
+        // Create a new user with mocked user and store their id
+        const author = await createUser(sonic);
+        uid = author._id;
+        console.log("Created the user");
+        // Create a new tuit by user with mocked tuit/user and store its id
+        testTuit = await createTuitByUser(author._id, mockedTuit);
+        tid = testTuit._id;
+        console.log("Created the tuit by user");
+    })
+
+    // Clean up after tests
+    afterAll(async () => {
+        // If the tuit had tags...
+        // if (testTuit.stats.dislikes > 0) {
+        //     // Remove the tag using the deleteTag function
+        //     await deleteTag(tagID);
+        // }
+
+        let promises = []   // Initialize an empty list of promises
+
+        // Add promise to delete users by username to list of promises
+        promises.push(deleteUsersByUsername(sonic.username));
+
+        // Add promise to delete tuits by tuit text to list of promises
+        promises.push(deleteTuitByTuitText(mockedTuit.tuit));
+
+        // Wait for and return the result of all promises
+        return Promise.all(promises);
+    })
+
+    test("user can create a tag in a tuit with REST API", async () => {
+        // Get all tags
+        const allTags = await findAllTags();
+
+        // Expect at least one tag
+        expect(allTags.length).toBeGreaterThanOrEqual(1);
+
+        // Expect #Sonic to be one of the tags
+        let flag = false;
+        for (let i = 0; i < allTags.length; i++) {
+            if (allTags[i].tag == 'Sonic') {
+                flag = true;
+            }
+        }
+        expect(flag).toEqual(true);
+
+    })
+
+});
+
+describe("can retrieve all tags with REST API", () => {
+    // Mocked tuits to insert
+    const mockedTuit1 = {
+        tuit: "Let's race, ace! #Sonic"
+    }
+
+    const mockedTuit2 = {
+        tuit: "I'm Sonic. Sonic the #hedgehog"
+    }
+    const mockedTuit3 = {
+        tuit: "I’m waaaaiiiitiiiing! #Sonic"
+    }
+
+    // Mocked user to insert
+    const sonic = {
+        username: 'SonicTheHedgehog',
+        password: 'chilidogs',
+        email: 'sonic@sega.com'
+    }
+
+    // Initialize test variables
+    let uid = null;
+    let tid1, testTuit1 = null;
+    let tid2, testTuit2 = null;
+    let tid3, testTuit3 = null;
+    let sonicID = null;
+
+    // Set up tests
+    beforeAll(async () => {
+        let promises = [];  // Initialize an empty list of promises
+
+        /*
+        Add promises to remove users with test username and
+        remove tuits with test texts to list of promises
+        */
+        promises.push(deleteUsersByUsername(sonic.username));
+        promises.push(deleteTuitByTuitText(mockedTuit1.tuit));
```

Comment:
These are missing await, similar to last PR.  Just need to get current version from main.